### ISSUE TITLE
Pattern Insertion Flows

### ISSUE THREAD

**Issue by priethor** [WordPress/gutenberg#31153]
### Insertion UI
- [x] #33046
- [x] #36697
- [x] #35364
- [x] #28312
- [x] #31033
  - Depends on: #28574
  - Precedes: #28736)
  - Deprecates: #26905
- [x] #26116
- [x] #27494
- [x] #20867
- [x] #40117

#### Pattern availability and restricting
- [x] #23275
- [x] #41398
- [x] #23803

#### Contextual suggestions (broader discussion in #27575)
- [x] #28735
- [ ] Block context: #28736,  #28309
  - [ ] #30631
- [ ] Container context: #28737
  - [ ] Size constraints: #30210 (e.g. avoid large pattern in small columns)
- [ ] Template context: #28739
- [x] Template part context: #28802

#### Search
- [x] #38536 

#### Connection with w.org directory
- [x] #28248
- [x] #28800

**Comment by ramonjd** [comment]
Looking for opinions here: would it make sense to be able to tell the block editor plugin [not to register its own patterns or update core patterns](https://github.com/WordPress/gutenberg/blob/trunk/lib/block-patterns.php#L245)? 

Let's just say we want to be able offer our own library of patterns, or only those available in a WordPress install. 

Other than keeping track of every block name that Gutenberg registers, and then manually unregistering them, how could we disable them?

I see two options (assuming the idea itself is sound):

1. Add an `apply_filters( 'register_gutenberg_patterns', true )` and `apply_filters( 'update_core_patternscheck', true )` for example
2. Standardize patterns names registered by the plugin so we can filter them out more reliably: currently we're registering `query/*` and `social-links/shared-background-color`.

I think it's especially important given the comment in the file:

```
 * Deactivate the legacy patterns bundled with WordPress, and add new block patterns for testing.
```

For experimental patterns, I think we should be able to opt-out.

**Comment by priethor** [comment]
Hi @ramonjd! Given this issue is more focused on the pattern insertion experience within post and template editors, I think it would be more effective to create its own issue to discuss this more in detail, as [suggested](https://github.com/Automattic/wp-calypso/pull/52656#discussion_r633194811) in the related PR. In any case, I think your proposal makes sense, but I would love to hear @ntsekouras' opinion as well.

**Comment by ramonjd** [comment]
Thanks for the quick feedback @priethor! I'll do as you suggest, and create a new issue to direct discussion there. Sorry for hijacking this issue!

**Comment by mtias** [comment]
@priethor I think we can close this tracking issue.
